include/in_mem_graph_store.h

```diff
@@ -11,13 +11,40 @@ namespace diskann
 class InMemGraphStore : public AbstractGraphStore
 {
   public:
-    InMemGraphStore(const size_t max_pts);
+    InMemGraphStore(const size_t max_pts, const size_t frozen_points);
 
     int load(const std::string &index_path_prefix);
-    int store(const std::string &index_path_prefix);
+    int store(const std::string &index_path_prefix, const size_t active_points);
 
-    void get_adj_list(const location_t i, std::vector<location_t> &neighbors);
-    void set_adj_list(const location_t i, std::vector<location_t> &neighbors);
+    virtual std::vector<location_t> &get_neighbours(const location_t i) override;
+    virtual void set_neighbours(const location_t i, std::vector<location_t> &neighbors) override;
+
+    virtual size_t resize_graph(const size_t new_size) override;
+    virtual void clear_graph() override;
+
+    virtual size_t get_num_frozen_points() override;
+    virtual size_t get_max_range_of_loaded_graph() override;
+    virtual uint32_t get_max_observed_degree() override;
+    virtual void set_max_observed_degree(uint32_t max_observed_degree) override;
+    virtual uint32_t get_start() override;
+    virtual void set_start(uint32_t start) override;
+
+    virtual size_t shrink_to_fit() override;
+
+  protected:
+    virtual location_t load_impl(const std::string &filename, size_t expected_num_points);
+#ifdef EXEC_ENV_OLS
+    virtual location_t load_impl(AlignedFileReader &reader, size_t expected_num_points);
+#endif
+
+    int save_graph(const std::string &index_path_prefix, const size_t active_points);
+
+  private:
+    size_t _max_range_of_loaded_graph = 0;
```

Comment:
whats the difference between max_range and max_observed_degree.

Reply:
there is no comment on main to explain this but this is for sure different from max_observed_degree